homeassistant/components/imeon_inverter/config_flow.py

```diff
@@ -0,0 +1,58 @@
+"""Config flow for Imeon integration."""
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ADDRESS, CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ImeonInverterConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle the initial setup flow for Imeon Inverters."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> config_entries.ConfigFlowResult:
+        """Handle the user step for creating a new configuration entry."""
+        schema = vol.Schema(
+            {
+                vol.Required("inverter"): str,
+                vol.Required(CONF_ADDRESS): str,
+                vol.Required(CONF_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+            }
+        )
+
+        if user_input is None:
+            return self.async_show_form(step_id="user", data_schema=schema)
+
+        # Check if entry already exists
+        existing_entry = self._find_existing_entry(
+            user_input[CONF_ADDRESS], user_input["inverter"]
+        )
+
+        if existing_entry:
```

Comment:
you'll need to find another way to check uniqueness and use
```
                    await self.async_set_unique_id(identifier)
                    self._abort_if_unique_id_configured()
```

Reply:
i can't figure out why my tests fail, all is working now that i added "
        await self.async_set_unique_id(user_input[CONF_ADDRESS])
        self._abort_if_unique_id_configured()
" but my tests give the error "
FAILED tests/components/imeon_inverter/test_config_flow.py::test_create_entry - TypeError: 'mappingproxy' object does not support item assignment
FAILED tests/components/imeon_inverter/test_config_flow.py::test_already_configured - TypeError: 'mappingproxy' object does not support item assignment
"
any ideas ?

Reply:
In fact, the method _abort_if_unique_id_configured() try to set a value "unique_id" to ConfigFlow.context["unique_id'] but context is not mutable, but I don't know how to deal with that because it comes from home assistant itself

Reply:
I'm not sure tbh. First thing I'd do was to rebase, to verify it's not something outdated.